src/Components/LoginForm.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react'
+import { Form, Button, Container, Message } from 'semantic-ui-react'
+import { connect } from 'react-redux'
+import { signInUser } from '../reduxTokenAuthConfig'
+
+class LoginForm extends Component {
+  state = {
+    email: '',
+    password: '',
+    message: '',
+    errorsLogin: ''
+  }
+
+  onChangeHandler = (e) => {
+    this.setState({
+      [e.target.id]: e.target.value
+    })
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    const { history, signInUser } = this.props
+    const {
+      email,
+      password
+    } = this.state
+    signInUser({ email, password })
+      .then(response => {
+        this.setState({ message: true })
+        setTimeout(function () { history.push('/') }, 3000)
+      }).catch(error => {
+        this.setState({
+          errorsLogin: error.response.data.errors[0],
+          message: false
+        })
+      })
+  }
+
+  render() {
+
+    let message
+    let user = this.props.currentUser.isSignedIn
+
+    if (user === true && this.state.message === true) {
```

Comment:
why do you need message state?

Reply:
I need it to display a `successful / or not` message to the user whenever he puts his credentials.

I did as you asked. Left only the `user`. On line 44.
And then when I was live testing i found out that:
if you try to login with invalid credentials you get the red message and if then you try to login with valid credentials the red message stays.
So it's more of a user experience thing.